### PR TITLE
Merge 'add/local-vip-go-env-const'... 

### PR DESCRIPTION
**Pull request by seanosh**
## Description

When setting up a local development environment that utilizes `vip-go-mu-plugins`, it's helpful to enable debug bar and relevant tooling. Setting `WPCOM_IS_VIP_ENV` to `false` is correct for a local environment, but doesn't offer a way to selectively enable something like debug tooling. 

I went with this approach because other logic for enabling mostly revolves around things specific to the platform (employee, proxy status) that would require additional/riskier changes or inclusion of hooks/filters at the onset locally, which isn't as "clean" as defining constant for storing environment type. 

cc: @joshbetz (thanks for being a sounding board on this!)

### Feature: Enable Logged Out Debug Mode for local envs

Enables Debug mode when `WP_ENVIRONMENT_TYPE' is defined and set to `local`:

``` 
define( 'WP_ENVIRONMENT_TYPE', 'local' );
```

## Checklist

Please make sure the items below have been covered before requesting a review:

- [ ] This change works and has been tested locally (or has an appropriate fallback).
- [ ] This change works and has been tested on a Go sandbox.
- [ ] This change has relevant unit tests (if applicable).
- [ ] This change has relevant documentation additions / updates (if applicable).
- [ ] I've created a changelog description that aligns with the provided examples. 

## Steps to Test
1. Check out and apply PR in local environment
2. Add to `wp-config.php`:
```
define( 'WPCOM_IS_VIP_ENV', false );
define( 'VIP_GO_ENV', 'local' );
define( 'VIP_BLOCK_WP_MAIL', true );
define( 'FILES_CLIENT_SITE_ID', 1 );
```
3. Visit local environment URL without having logged in and observe debug panel presence.

### CODE REVIEW

**Comment by mjangda** [comment]
Might be better to set and check `'local' === WP_ENVIRONMENT_TYPE`? (https://make.wordpress.org/core/2020/07/24/new-wp_get_environment_type-function-in-wordpress-5-5/)

A non-false value for VIP_GO_ENV might have other unintended consequences.

**Comment by seanosh** [comment]
Nice, thanks @mjangda - updated the PR accordingly.

**Comment by rinatkhaziev** [comment]
To really make it magical we also need to define the `local` in this block: 

https://github.com/Automattic/vip-go-mu-plugins/blob/02729f0e4f23c72dee12d2b2208d9b2384e42fd7/000-vip-init.php#L208-L224

File: 000-debug/debug-mode.php [hunk 1]
```diff
@@ -39,8 +39,9 @@ function is_debug_mode_enabled() {
 	$is_nocache = isset( $_COOKIE['vip-go-cb'] ) && '1' === $_COOKIE['vip-go-cb'];
 	$is_debug = isset( $_COOKIE['a8c-debug'] ) && '1' === $_COOKIE['a8c-debug'];
 	$is_proxied = \is_proxied_request();
+	$is_local = ( ( defined( 'WP_ENVIRONMENT_TYPE' ) && 'local' === WP_ENVIRONMENT_TYPE ) ? true : false );
```
Comment (rinatkhaziev) [inline 000-debug/debug-mode.php:42]:
```suggestion
	$is_local = defined( 'WP_ENVIRONMENT_TYPE' ) && 'local' === WP_ENVIRONMENT_TYPE;
```

Comment (rinatkhaziev) [reply 000-debug/debug-mode.php:42]: The ternary is redundant here.

**Review by rinatkhaziev — CHANGES_REQUESTED** [review]
Thanks for the addition @seanosh, left a suggestion.

**Review by rinatkhaziev — APPROVED** [review]

**Comment by rinatkhaziev** [comment]
Updated changelog entry

**Comment by nickdaugherty** [comment]
r1443-stacks

**Comment by nickdaugherty** [comment]
Release v1.22.0

**Pull request merged**